boa/src/value/conversions.rs

```diff
@@ -186,3 +186,9 @@ where
         }
     }
 }
+
+impl From<std::num::TryFromIntError> for Value {
+    fn from(err: std::num::TryFromIntError) -> Self {
+        Value::string(format!("{}", err))
+    }
+}
```

Comment:
Why would we need to convert a `TryFromIntError` to a `Value::String`?

Reply:
`Value` is used to represent errors in most of the array-related functions and, in this PR, I use `i*::try_from(...)?` a lot there. I don't think there is a better way to return this error without adding another `Value` case or switching to non-`Value` errors (or mapping everything manually on each call site).

Reply:
Hmmm I don't think we should add this to `Value`. We can use `map_err()` in the error being returned by the integer conversion, and then create a value from it, but wherever this is being used. `Value` should not be aware that we always want to convert an invalid integer conversion to a string, because we might want to convert it to an `Error` at some other point.